scripts/compile_parsers.makefile

```diff
@@ -1,18 +1,51 @@
 #
 # compile_parsers.makefile
-# Stephan Seitz, 2021-09-09 21:36
 #
-CC?=cc
-CXX_STANDARD?=c++14
-C_STANDARD?=c99
 
-all: parser.so
+CFLAGS       ?= -std=c99 -fPIC
+CXXFLAGS     ?= -std=c++14 -fPIC
+LDFLAGS      ?= -Os -shared
+SRC_DIR      ?= ./src
+DEST_DIR     ?= ./dest
 
-parser.o: src/parser.c
-	$(CC) -c src/parser.c -std=$(C_STANDARD) -fPIC -I./src
+ifeq ($(OS),Windows_NT)
+   SHELL       := powershell.exe
+   .SHELLFLAGS := -NoProfile
+   RM          := Remove-Item -Force
+   CP          := Copy-Item -Recurse
+   MKDIR       := New-Item -ItemType directory
+   TARGET      := parser.dll
+else
+   RM          := rm -rf
+   CP          := cp
+   TARGET      := parser.so
+endif
 
-scanner.o: src/scanner.cc
-	$(CC) -c src/scanner.cc -std=$(CXX_STANDARD) -fPIC -I./src
+ifneq ($(wildcard src/*.cc),)
+   LDFLAGS += -lstdc++
+endif
 
-parser.so: parser.o scanner.o
-	$(CC) parser.o scanner.o -o parser.so -shared -Os -lstdc++
+OBJECTS   := parser.o scanner.o
+
+all: $(TARGET)
+
+$(TARGET): $(OBJECTS)
+	$(CC) $(OBJECTS) -o $(TARGET) $(LDFLAGS)
+
+%.o: src/%.c
```

Comment:
@kylo252 wouldn't it be better to go back to `parser.c` and `scanner.{c,cc}`. Those are the only filenames supported by upstream tree-sitter (the filenames are hard-coded into the Rust code of tree-sitter CLI. I don't know whether we want to glob in just any C file in that folder.

Reply:
I'm not sure if it's possible without doing the glob/wildcard at some point in the recipe. The `scanner.cc` file needs to exist if we were to add it as an explicit dep for one of the targets.

Any ideas how to get around that?